### PR TITLE
Provide details for configuring Spark Integration with Datadog Agent on DC/OS

### PR DESCRIPTION
**Pull request by xornivore**
[Rendered](https://github.com/xornivore/datadog-agent-dcos-ee-strict/blob/xornivore/spark-config-details/README.md)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -49,48 +56,58 @@ $ tree ./opt-datadog-agent
 │   ├── mesos_master.d
 │   │   ├── conf.yaml
 │   │   └── private-key.pem
-│   └── mesos_slave.d
+│   ├── mesos_slave.d
+│   │   ├── conf.yaml
+│   │   └── private-key.pem
```
Comment (clamoriniere) [inline README.md:61]: maybe you can remove the `private-key.pem` from the repo.

File: README.md [hunk 2]
```diff
@@ -49,48 +56,58 @@ $ tree ./opt-datadog-agent
...
+│   └── spark.d
 │       ├── conf.yaml
 │       └── private-key.pem
 └── cont-init.d
     └── 90-mesos.sh
 ```
 
-Please make sure to replace the `private-key.pem` file with the contents of the private key generated for the service account.
+Please make sure to replace the `private-key.pem` file with the contents of the private key generated for the service
```
Comment (clamoriniere) [inline README.md:69]:
```suggestion
Please make sure to copy the private key `private-key.pem` generated for the service account, in each integration folder: `mesos_master.d`, `mesos_slave.d` and `spark.d`.
```

File: README.md [hunk 3]
```diff
@@ -178,3 +193,217 @@ Agent (v7.24.0-devel+git.14.c805c57)
...
         version.scheme: semver
 ```
+
+## Spark Configuration
+
+In order for the Datadog Agent Spark integration to be able to collect metrics in the DC/OS running in Strict security
+mode the Spark and Spark History Server packages have to be configured according to the D2IQ security recommendations
+for this mode of operation.
+
+The following instructions are provided as a reference only. If you have already installed Spark and History Server you
```
Comment (clamoriniere) [inline README.md:203]:
```suggestion
> :warning:  The following instructions are provided as a reference only. If you have already installed Spark and History Server you
```

File: deploy/hdfsclient.json [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+      {
+        "mode": "host"
+      }
+    ],
+    "env": {
+      "JAVA_HOME": "/usr/lib/jvm/default-java",
+      "HDFS_SERVICE_NAME": "hdfs"
+    }
+  }
+
```
Comment (clamoriniere) [inline deploy/hdfsclient.json:23]:
```suggestion
```

File: deploy/spark-history-options.json [hunk 5]
```diff
@@ -0,0 +1,7 @@
+{
+    "service": {
+        "hdfs-config-url": "http://api.hdfs.marathon.l4lb.thisdcos.directory/v1/endpoints",
+        "service_account": "spark",
+        "service_account_secret": "spark/sa-secret"
+    }
+}
```
Comment (clamoriniere) [inline deploy/spark-history-options.json:7]:
```suggestion
}

```

File: deploy/spark-options.json [hunk 6]
```diff
@@ -0,0 +1,10 @@
+{
+    "service": {
+      "service_account": "spark",
+      "service_account_secret": "spark/sa-secret",
+      "spark-history-server-url": "https://leader.mesos/service/spark-history"
+    },
+    "hdfs": {
+      "config-url": "http://api.hdfs.marathon.l4lb.thisdcos.directory/v1/endpoints"
+    }
+  }
```
Comment (clamoriniere) [inline deploy/spark-options.json:10]:
```suggestion
  }

```

File: opt-datadog-agent/conf.d/spark.d/conf.yaml [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+      reader:
+        type: dcos_auth
+        login_url: https://leader.mesos/acs/api/v1/auth/login
+        service_account: datadog_agent
+        private_key_path: /conf.d/spark.d/private-key.pem
+        expiration: 600 # 10 min
+      writer:
+        type: header
+        name: Authorization
+        value: token=<TOKEN>
```
Comment (clamoriniere) [inline opt-datadog-agent/conf.d/spark.d/conf.yaml:20]:
```suggestion
        value: token=<TOKEN>

```

**Review by clamoriniere — APPROVED** [review]
few comments, but look good 👍

File: README.md (hunk 1 above)
Comment (xornivore) [reply README.md:61]: it's an empty key - I left it to make sure it's clear we need one, but it's a placeholder here.

**Pull request merged**